app/src/main/java/ch/hikemate/app/ui/map/MapScreen.kt

```diff
@@ -0,0 +1,271 @@
+    // time?
+    addMapListener(
+        object : MapListener {
+          override fun onScroll(event: ScrollEvent?): Boolean {
+            hikingRoutesViewModel.setArea(mapView.boundingBox)
+            return true
+          }
+
+          override fun onZoom(event: ZoomEvent?): Boolean {
+            hikingRoutesViewModel.setArea(mapView.boundingBox)
+            return true
+          }
+        })
+  }
+
+  Box(modifier = Modifier.fillMaxSize()) {
+    AndroidView(
+        factory = { mapView }, modifier = Modifier.fillMaxSize().testTag(MapScreen.TEST_TAG_MAP))
+
+    IconButton(
+        onClick = {
+          // TODO : Adapt the map screen to navigation
+          Toast.makeText(context, "Menu not implemented yet", Toast.LENGTH_SHORT).show()
+        },
+        modifier =
+            Modifier.padding(16.dp)
+                .align(Alignment.TopStart)
+                // Clip needs to be before background
+                .clip(RoundedCornerShape(8.dp))
+                .background(MaterialTheme.colorScheme.surface)
+                .testTag(MapScreen.TEST_TAG_MENU_BUTTON)) {
+          Icon(
+              Icons.Default.Menu,
+              contentDescription =
+                  context.getString(R.string.map_screen_menu_button_content_description),
+              tint = MaterialTheme.colorScheme.onSurface)
+        }
+
+    CollapsibleHikesList(hikingRoutesViewModel)
+  }
+
+  // Initialize the list of hiking routes once when the map is loaded
+  // Leave subsequent updates to the map listener
+  LaunchedEffect(Unit) { hikingRoutesViewModel.setArea(mapView.boundingBox) }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CollapsibleHikesList(hikingRoutesViewModel: ListOfHikeRoutesViewModel) {
+  val scaffoldState = rememberBottomSheetScaffoldState()
+  val routes = hikingRoutesViewModel.hikeRoutes.collectAsState()
+  val context = LocalContext.current
+
+  BottomSheetScaffold(
+      scaffoldState = scaffoldState,
+      sheetContainerColor = MaterialTheme.colorScheme.surface,
+      sheetContent = {
+        Column(modifier = Modifier.fillMaxSize().testTag(MapScreen.TEST_TAG_HIKES_LIST)) {
+          LazyColumn(modifier = Modifier.fillMaxSize()) {
+            if (routes.value.isEmpty()) {
+              item {
+                // Use a box to center the Text composable of the empty list message
+                Box(modifier = Modifier.fillMaxWidth(), contentAlignment = Alignment.Center) {
+                  Text(
+                      text = context.getString(R.string.map_screen_empty_hikes_list_message),
+                      style = MaterialTheme.typography.bodyLarge,
+                      // Align the text within the Text composable to the center
+                      textAlign = TextAlign.Center)
+                }
+              }
+            } else {
+              items(routes.value.size) { index: Int ->
+                HikingRouteItem(
+                    title = routes.value[index],
```

Comment:
Note: Consider add a comment here that there values are hardcoded and will be updated with real values later

Reply:
I created [this draft task](https://github.com/orgs/HikeMate/projects/1/views/2?pane=issue&itemId=83421220) for the altitude difference. `isSuitable` would require more work from what I gather, it would rather be a user story